aii-core/src/main/perl/Shellfe.pm

```diff
@@ -323,6 +339,38 @@ sub _initialize
     if ($self->option(INCLUDE)) {
         unshift(@INC, split(/:+/, $self->option(INCLUDE)));
     }
+
+    # Initialization and validation of some options whose default values are based on other options
+    if ( $self->option(NBPDIR_PXELINUX) ) {
+        $self->{CONFIG}->set(NBPDIR_PXELINUX, undef) if $self->option(NBPDIR_PXELINUX) eq NBPDIR_VARIANT_DISABLED;
+    }
+    if ( $self->option(NBPDIR_GRUB2) ) {
+        $self->{CONFIG}->set(NBPDIR_GRUB2, undef) if $self->option(NBPDIR_GRUB2) eq NBPDIR_VARIANT_DISABLED;
```

Comment:
Note sure to understand what you suggest... here the goal is to undefine the option if it happens to be defined as Grub2 support is not activated... (NBPDIR_VARIANT_DISABLED)